collectors/ebpf.plugin/ebpf_socket.h

```diff
@@ -118,8 +138,72 @@ typedef struct ebpf_network_viewer_options {
 
     ebpf_network_viewer_hostname_list_t *excluded_hostnames;
     ebpf_network_viewer_hostname_list_t *included_hostnames;
+
+    ebpf_network_viewer_ip_list_t *ipv4_local_ip;
+    ebpf_network_viewer_ip_list_t *ipv6_local_ip;
 } ebpf_network_viewer_options_t;
 
 extern ebpf_network_viewer_options_t network_viewer_opt;
 
+/**
+ * Structure to store socket information
+ */
+typedef struct netdata_socket {
+    uint64_t recv_packets;
+    uint64_t sent_packets;
+    uint64_t recv_bytes;
+    uint64_t sent_bytes;
+    uint64_t first; //First timestamp
+    uint64_t ct;   //Current timestamp
+    uint16_t retransmit; //It is never used with UDP
+    uint8_t protocol; //Should this to be in the index?
+    uint8_t removeme;
+    uint32_t reserved;
+} netdata_socket_t __attribute__((__aligned__(8)));
+
+/**
+ * Index used together previous structure
+ */
+typedef struct netdata_socket_idx {
+    union netdata_ip_t saddr;
+    union netdata_ip_t daddr;
+    uint16_t dport;
+    uint16_t sport;
```

Comment:
```suggestion
    uint16_t sport;
    uint16_t dport;
```

Reply:
For we do this inversion here, it will be necessary to invert also in the ebpf program:

https://github.com/netdata/kernel-collector/blob/6a598c67312917493be7ef1b41048ce0c3dc5f8d/kernel/network_viewer_kern.c#L56